### ISSUE TITLE
[MJAVADOC-528] Invalid 'expires' attribute

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#838]
**[Roberto Benedetti](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=roberto.benedetti)** opened **[MJAVADOC-528](https://issues.apache.org/jira/browse/MJAVADOC-528?redirect=false)** and commented

When detectLinks is enabled and the plugin checks if the url is valid, warnings like this

```java
Invalid cookie header: "Set-Cookie: logged_in=no; domain=.github.com; path=/; expires=Sun, 06 Jun 2038 14:02:43 -0000; secure; HttpOnly". Invalid 'expires' attribute: Sun, 06 Jun 2038 14:02:43 -0000
```

may appear.

Previous warning was due to `com.google.code.gson:gson:jar:2.5` dependency and is raised by HttpClient which does not expect 'expires' values compliant to RFC 6265.
The attachment is a patch which makes HttpClient ignore cookies. Maybe a better solution would be upgrading to the latest HttpComponents API which supports RFC 6265.

---

**Affects:** 3.0.1

**Attachments:**
- [cookie.patch](https://issues.apache.org/jira/secure/attachment/12926739/cookie.patch) (_1.23 kB_)

**Remote Links:**
- [GitHub Pull Request #141
](https://github.com/apache/maven-javadoc-plugin/pull/141)
- [GitHub Pull Request #5
](https://github.com/apache/maven-javadoc-plugin/pull/5)

0 votes, 6 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I am inclined to drop the entire line. Have you tried that?

**Comment by jira-importer** [comment]
**[Roberto Benedetti](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=roberto.benedetti)** commented

Would do you mean?

I added the line to ignore cookies because HttpClient parses them by default and expects an old format format for 'expires'.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

The entire config line should be dropped. No cookie config.

**Comment by jira-importer** [comment]
**[Roberto Benedetti](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=roberto.benedetti)** commented

I created a PR. Without cookie config HttpClient defaults to `CookiePolicy.BEST_MATCH` which is affected by the issue.
Instead of disabling cookies the PR sets the policy to `CookiePolicy.BROWSER_COMPATIBILITY` which works fine.

**Comment by jira-importer** [comment]
**[Guillaume Boué](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gboue)** commented

Fixed in [b651443bba06fdf469b730024489c3f02629e50c](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git;a=commit;h=b651443bba06fdf469b730024489c3f02629e50c).
